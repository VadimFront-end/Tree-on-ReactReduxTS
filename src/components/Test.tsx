import React from 'react';

interface IProps {
    name: string,
    id: number
}

const Test = ({name, id}: IProps) => {
    return (
        <div className="Test">
            {
                id ? <div>{name}</div> : <div>fuck</div>
            }
        </div>
    );
}

export default Test;
