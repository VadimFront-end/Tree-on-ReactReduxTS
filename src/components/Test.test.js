import React from 'react';
import renderer from 'react-test-renderer';
import Test from './Test';
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter() });

const setUp = props => shallow(<Test {...props}/>);

describe('Testing component `Test`', () => {
    it('Can i test component with Snapshot', () => {
        const component = renderer.create(
            <Test name={'Vlad'} id={1}/>
        );
        expect(component).toMatchSnapshot();
    })

    it('Can i test component with enzyme', () => {
        const component = setUp({name: 'Vadim', id: 1});
        console.log(component.debug())
        expect(component.find('.Test').length).toBe(1);
    });
})
