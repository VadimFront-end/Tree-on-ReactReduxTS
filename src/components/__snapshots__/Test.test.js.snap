// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component \`Test\` Can i test component with Snapshot 1`] = `
<div
  className="Test"
>
  <div>
    Vlad
  </div>
</div>
`;
